### ISSUE TITLE
Investigate moving code related to block templates in class-wc-template-loader.php to BockTemplatesController.php

### ISSUE THREAD

**Issue by Aljullu** [woothemes/woocommerce#44540]
In `class-wc-template-loader.php` there is some code that determines if there is a block template available and, based on that, enables or disables the classic template routing:

https://github.com/woocommerce/woocommerce/blob/143388dae8507ce1c749e2a6d53e8b1115747539/plugins/woocommerce/includes/class-wc-template-loader.php#L174-L215

Ideally, all code related to block templates should live in `BlockTemplatesController.php`.

This issue is about investigating how much of that code can be moved.

**Comment by Aljullu** [comment]
~In parallel to this, we need to investigate if we can clean up some `template_redirect` actions we have in block templates code which are only used to set `woocommerce_has_block_template` to `__return_true`.~ Update: this was already completed with the class refactor PR: https://github.com/woocommerce/woocommerce/pull/44537.

**Comment by Aljullu** [comment]
I'm going to close this issue as there is nothing actionable and I can't see any obvious ways to move this code to BlockTemplatesController considering that it will only be initialized in block themes (see #48905).

We have a separate issue about improving the logic in `class-wc-template-loader.php` which I think is the best path forward: https://github.com/woocommerce/woocommerce/issues/47887.
